lib/os/mem_blocks.c

```diff
@@ -364,15 +365,15 @@ int sys_multi_mem_blocks_alloc(sys_multi_mem_blocks_t *group,
 		goto out;
 	}
 
-	if (count > allocator->num_blocks) {
+	if (count > allocator->info.num_blocks) {
 		ret = -ENOMEM;
 		goto out;
 	}
 
 	ret = sys_mem_blocks_alloc(allocator, count, out_blocks);
 
 	if ((ret == 0) && (blk_size != NULL)) {
-		*blk_size = BIT(allocator->blk_sz_shift);
+		*blk_size = BIT(allocator->info.blk_sz_shift);
```

Comment:
I guess this and previous changes actually belong to the "refactor sys_mem_blocks" patch.

Reply:
The perils of continuous refactoring ... Fixing.